client/web/src/components/popovers/ColorPicker.vue

```diff
@@ -115,13 +117,197 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
+import { clamp, HSV2RGB, RGB2HSV, RGB, HSV } from "../../lib/utils";
+
+interface Rect {
+	width: number;
+	height: number;
+	left: number;
+	top: number;
+}
+
+interface ColorPickerData {
+	color: HSV;
+	hue: {
+		rect: Rect;
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
+
+type ColorPickerState = "idle" | "move_hue" | "move_opacity" | "move_saturation";
 
 export default defineComponent({
+	colorPicker: {
```

Comment:
Is there a reason this is a custom property instead of part of the `data` section?

Reply:
This data does not need to be reactive.

Reply:
https://vuejs.org/v2/api/#vm-options

Reply:
Alternatively, we can put the object in data() and prefix with an `_`, VueJS does not proxify object with underscore. 
https://v3.vuejs.org/api/options-data.html#data-2